h2/src/main/org/h2/command/Parser.java

```diff
@@ -4721,7 +4721,7 @@ private Expression readTerm() {
                     read("PG_CATALOG");
                     read(DOT);
                 }
-                if (readIf("REGCLASS")) {
+                if (readIf("REGCLASS") || readIf("OID")) {
```

Comment:
`::regclass` and `::oid` are different things in PostgreSQL and this additional condition doesn't look like a proper implementation.

H2 has an incorrect mapping for `OID` data type to a `BLOB` in `DataType` class. I think this mapping should be removed from it (and from `help.csv`), and a new mapping to `INTEGER` should be created in the `Mode` class.

```Java
dt = DataType.getDataType(Value.INTEGER);
mode.typeByNameMap.put("OID", dt);
```
With such change `::oid` will be processed just like any other type cast. Currently this cast is broken, because it tries to cast a numeric value to a `BLOB`.

Reply:
This really works :-)